src/plugins/chart_expressions/expression_metric/public/components/metric_vis.tsx

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback } from 'react';
+
+import numeral from '@elastic/numeral';
+import { i18n } from '@kbn/i18n';
+import { Chart, Metric, MetricSpec, RenderChangeListener, Settings } from '@elastic/charts';
+import { getColumnByAccessor, getFormatByAccessor } from '@kbn/visualizations-plugin/common/utils';
+import { ExpressionValueVisDimension } from '@kbn/visualizations-plugin/common';
+import { Datatable, IInterpreterRenderHandlers } from '@kbn/expressions-plugin';
+import { CustomPaletteState } from '@kbn/charts-plugin/public';
+import { euiLightVars } from '@kbn/ui-theme';
+import { VisParams } from '../../common';
+import { getPaletteService, getThemeService, getFormatService } from '../services';
+import { currencyCodeMap } from './currency_code_map';
+
+const defaultColor = euiLightVars.euiColorDarkestShade;
+
+const getBytesUnit = (value: number) => {
+  const units = ['byte', 'kilobyte', 'megabyte', 'gigabyte', 'terabyte', 'petabyte'];
+  const abs = Math.abs(value);
+
+  const base = 1024;
+  let unit = units[0];
+  let matched = abs < base;
+  let power;
+
+  if (!matched) {
+    for (power = 1; power < units.length; power++) {
+      const [min, max] = [Math.pow(base, power), Math.pow(base, power + 1)];
+      if (abs >= min && abs < max) {
+        unit = units[power];
+        matched = true;
+        value = value / min;
+        break;
+      }
+    }
+  }
+
+  if (!matched) {
+    value = value / Math.pow(base, units.length - 1);
+    unit = units[units.length - 1];
+  }
+
+  return { value, unit };
+};
+
+const getFormatter = (
+  accessor: ExpressionValueVisDimension | string,
+  columns: Datatable['columns']
+) => {
+  const serializedFieldFormat = getFormatByAccessor(accessor, columns);
+  const formatId = serializedFieldFormat?.id ?? 'number';
+
+  if (!['number', 'currency', 'percent', 'bytes', 'duration'].includes(formatId)) {
+    throw new Error(
+      i18n.translate('expressionMetricVis.errors.unsupportedColumnFormat', {
+        defaultMessage: 'Metric Visualization - Unsupported column format: "{id}"',
+        values: {
+          id: formatId,
+        },
+      })
+    );
+  }
+
+  if (formatId === 'duration') {
+    const formatter = getFormatService().deserialize({
+      ...serializedFieldFormat,
+      params: {
+        ...serializedFieldFormat!.params,
+        outputFormat: 'humanizePrecise',
+        outputPrecision: 1,
+        useShortSuffix: true,
+      },
+    });
+    return formatter.getConverterFor('text');
+  }
+
+  const locale = String(numeral.language());
```

Comment:
This is super weird, but it seems like the numeral language isn't set correctly before at least one number got formatted: https://github.com/elastic/kibana/blob/d6badd8715ecf37fb7899f85a671be48031e67bd/src/plugins/field_formats/common/converters/numeral.ts#L45

We should probably get it via uiSettings and `FORMATS_UI_SETTINGS.FORMAT_NUMBER_DEFAULT_LOCALE` to be safe

Reply:
I guess I will have to call `numeral.language(locale)` to get the correct `languageData` loaded. I don't love it, but it does seem to be the most direct fix.